splunk-otel-android/src/main/java/io/opentelemetry/rum/internal/instrumentation/anr/AnrDetectorToggle.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.rum.internal.instrumentation.anr;
+
+import androidx.annotation.Nullable;
+import io.opentelemetry.rum.internal.instrumentation.ApplicationStateListener;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+final class AnrDetectorToggle implements ApplicationStateListener {
```

Comment:
Consider renaming to `AnrWatcherToggler`? I think it's a little more accurate. Also "toggle" as a noun doesn't really fit here, so "toggler" or "initiator" or something.